src/index.ts

```diff
@@ -4,6 +4,14 @@ import type { Compiler } from 'webpack';
 
 let inode = 45000000;
 
+let FS_ACCURACY = 2000;
+function applyMtime (mtime) {
+    if (FS_ACCURACY > 1 && mtime % 2 !== 0) FS_ACCURACY = 1;
+    else if (FS_ACCURACY > 10 && mtime % 20 !== 0) FS_ACCURACY = 10;
+    else if (FS_ACCURACY > 100 && mtime % 200 !== 0) FS_ACCURACY = 100;
+    else if (FS_ACCURACY > 1000 && mtime % 2000 !== 0) FS_ACCURACY = 1000;
+};
+
```

Comment:
This is an assumption but a quick google of "filesystem precision" shows that filesystems have differing amounts of timestamp precision. 

[Check this thread](https://unix.stackexchange.com/a/490990)
> The things I learned so far were:
> 
> - XFS and EXT3: second precision
> - EXT4: millisecond precision
> - NTFS: 100ns precision (OK, got this from the docs...)

So I figure this is addressed to that fact.

Reply:
Yes, but it does not explain all the four values, so you are on the right way, but the explanation is still incomplete